openmc/tallies.py

```diff
@@ -1653,7 +1659,8 @@ def _align_tally_data(self, other):
             if self_repeat_factor == 1:
```

Comment:
To be honest, I forget why exactly, but I know (think) there is a good reason for it. You might consider creating some toy test tallies and manipulate them with tally arithmetic in an ipython notebook. In particular, create tallies + arithmetic operation which cycle through these different `if-else` branches to see if the behavior makes sense. Not that it is necessary for this PR, but our test suite could include a "test-tally-arithmetic" test to do this sort of thing to maximize code coverage (since the `openmc.mgxs` tests only cover some portions of the tally arithmetic code).